apidump/net.go

```diff
@@ -43,26 +45,51 @@ func showPermissionErrors(sampleError error) error {
 			if env.InDocker() {
 				printer.Warningf("Although you are running as root, this container lacks the CAP_NET_RAW capability.\n")
 				printer.Warningf("It might be that you are in a PaaS that disallows packet capture, or the local configuration has disabled that privilege by default.\n")
-				return errors.Errorf("Insufficient permissions in container.")
+				return NewApidumpError(api_schema.ApidumpError_PCAPPermission, "Insufficient permissions in container.")
 			} else {
 				printer.Warningf("Although you are running as root, the Akita agent lacks the CAP_NET_RAW capability.\n")
 				printer.Warningf("It might be that you are in a restricted environment which disallows packet capture, even as the root user.\n")
-				return errors.Errorf("Insufficient permissions.")
+				return NewApidumpError(api_schema.ApidumpError_PCAPPermission, "Insufficient permissions.")
 			}
 		} else {
 			// Non-root user
 			printer.Warningf("The agent needs the CAP_NET_RAW capability to capture packets. You are running as an unprivileged (non-root) user.\n")
-			return errors.Errorf("Insufficient permissions, try using \"sudo\" to run as root.")
+			return NewApidumpError(api_schema.ApidumpError_PCAPPermission, "Insufficient permissions, try using \"sudo\" to run as root.")
+		}
+	} else if strings.Contains(sampleError.Error(), "SIOCETHTOOL(ETHTOOL_GET_TS_INFO) ioctl failed: Function not implemented") {
+		// This happens when the binary was built for a different architecture, e.g.
+		// if the user pulled the amd64 Docker image on arm64.
+		arch := architecture.GetCanonicalArch()
+
+		printer.Warningf(
+			"The agent received \"Function not implemented\" when trying to read from your network interfaces. "+
+				"This often indicates that the Akita agent was built for a different architecture than your host architecture. "+
+				"This Akita agent binary was built for %s.\n",
+			arch,
+		)
+
+		if env.InDocker() {
+			return errors.Errorf(
```

Comment:
Missing NewApidumpError here?  Will take a longer look later.

Reply:
Whoops, thanks.  Fixed in a4cf4764fb2a61494fbbb60af14620c089ede27e.